include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -70,34 +70,24 @@ namespace dht
 		void reset();
 	};
 
-	struct TORRENT_EXPORT dht_immutable_item
+	struct TORRENT_EXPORT dht_immutable_data
 	{
-		dht_immutable_item() = default;
-		dht_immutable_item(dht_immutable_item const& item);
-		dht_immutable_item(dht_immutable_item&&) = default;
-		dht_immutable_item& operator=(dht_immutable_item const& item);
-		dht_immutable_item& operator=(dht_immutable_item&&) = default;
-
 		// the actual value
-		std::unique_ptr<char[]> value;
+		std::vector<char> value;
 		// this counts the number of IPs we have seen
 		// announcing this item, this is used to determine
 		// popularity if we reach the limit of items to store
-		bloom_filter<128> ips;
+		bloom_filter<128> popularity;
 		// the last time we heard about this item
 		// the correct interpretation of this field
 		// requires a time reference
 		time_point last_seen;
```

Comment:
another benefit of having separate types is that this timestamp could be a ``time_t`` proper, and you wouldn't need the ``timestamp`` field below.